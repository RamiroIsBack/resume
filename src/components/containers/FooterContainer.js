import React, {Component} from 'react';
import {Footer_css} from '../../css';
import {connect} from 'react-redux';
import actions from '../../actions';
import {PopularFunctions} from '../../utils';
import {MapaContainer} from './'

class FooterContainer extends Component {

  handleClick(event){
    window.open(event.target.id,'_blank');
  }

  render() {
    let gMapsCopy = {};
    let emailCopy = {};
    let telCopy = {};
    let downloadCopy = {};
    let connectList = [];
    let footerList =[];
    if(this.props.copy){
      if (this.props.copy.footerList.length != 0){
        footerList = this.props.copy.footerList;
        for(let i = 0; i<footerList.length; i++){
          if(footerList[i].nombre === 'download'){
            downloadCopy = footerList[i];
          }
          if(footerList[i].nombre === 'gMaps'){
            gMapsCopy = footerList[i];
          }
          if(footerList[i].nombre === 'email'){
            emailCopy = footerList[i];
          }
          if(footerList[i].nombre === 'tel'){
            telCopy = footerList[i];
          }else if (footerList[i].nombre === 'instagram' ||
            footerList[i].nombre === 'stackOverFlow' ||
            footerList[i].nombre === 'linkedIn' ||
            footerList[i].nombre === 'gitHub'){
            //meter en el connectList
            connectList.push(
              <div key ={i}
                onClick = {this.handleClick.bind(this)}
                id = {footerList[i].urlLink}
              >
                <img className = 'footer__connect__foto'
                  src = {footerList[i].urlPic}
                  id = {footerList[i].urlLink}
                ></img>

              </div>
            );
          }
        }
      }
    }

    let visual = PopularFunctions.figureOutOpacity(this.props);

    return (
      <div className='footer__container' style = {visual}>
        <div className='footer__map__container'>
          <MapaContainer
            mapInfo = {gMapsCopy}
            googleMapURL='https://maps.googleapis.com/maps/api/js?key=AIzaSyClcb4B5oRktWDQWGU8Ev4hgYm5p_NXgL4&v=3.exp&libraries=geometry,drawing,places'
            loadingElement={<div style={{ height: '100%' }} />}
          />
        </div>
        <div className='footer__tel__container'>
          <div className='footer__tel'>
            <a href={'tel:' + telCopy.urlLink} style={{color: 'white',fontWeight:'bold'}}>{telCopy.urlLink}</a>
          </div>
          <div className='footer__tel__pic'>
            <img className='footer__picPhoto'
              src= {telCopy.urlPic}
            ></img>
          </div>
        </div>
        <div className='footer__mail__container'>
          <div className='footer__mail'>
            <a href={'mailto:' + emailCopy.urlLink} style={{color: 'white',fontWeight:'bold'}}>{emailCopy.urlLink}</a>

          </div>
          <div className='footer__mail__pic'>
            <img className='footer__picPhoto'
              src= {emailCopy.urlPic}
            ></img>
          </div>
        </div>

        <div className='footer__connect__foto__container'>
          {connectList}
        </div>
        <div className='footer__download__foto__container'>
          <a className = 'footer__download__text__link__container'
            href={downloadCopy.urlLink}
            download='Ramiro Santamaria Resume.pdf'>
            <h4 className ='footer__download__text'>Download the one-page .pdf</h4>
          </a>
          <a className = 'footer__download__foto__link__container'
            href={downloadCopy.urlLink}
            download='Ramiro Santamaria Resume.pdf'>
            <img className='footer__download__foto'
              src= {downloadCopy.urlPic}
              alt= 'download' >
            </img>
          </a>
        </div>

      </div>
    );
  }
}

const dispatchToProps = (dispatch) =>{

  return{
    getCopy: () => dispatch(actions.getCopy()),
    imgLoaded: (loaded) => dispatch (actions.imgLoaded(loaded)),

  };
};
const stateToProps = (state) => {
  return{
    copy :state.copy,
    section: state.section,
  };
};

export default connect (stateToProps,dispatchToProps)(FooterContainer);
