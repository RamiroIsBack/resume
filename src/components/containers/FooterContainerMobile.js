import React, {Component} from 'react';
import {Footer_css} from '../../css';
import {connect} from 'react-redux';
import actions from '../../actions';
import {PopularFunctions} from '../../utils';

class FooterContainerMobile extends Component {


  render() {
    let urlPic ='';

    let copy = PopularFunctions.selectSpecificCopy(this.props,'footer');
    let visual = PopularFunctions.figureOutOpacity(this.props);

    return (
      <div className='footer__container' style = {visual}>

      </div>
    );
  }
}

const dispatchToProps = (dispatch) =>{

  return{
    getCopy: () => dispatch(actions.getCopy()),
    imgLoaded: (loaded) => dispatch (actions.imgLoaded(loaded)),

  };
};
const stateToProps = (state) => {
  return{
    copy :state.copy,
    section: state.section,
  };
};

export default connect (stateToProps,dispatchToProps)(FooterContainerMobile);
