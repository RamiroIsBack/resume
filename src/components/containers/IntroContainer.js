import React, {Component} from 'react';
import {Intro_css} from '../../css';
import {connect} from 'react-redux';
import actions from '../../actions';
import {PopularFunctions} from '../../utils';

class IntroContainer extends Component {
  componentDidMount(){
    if (!this.props.copy.copyLoaded){
      this.props.getCopy();
    }
  }
  handleImageLoaded(){

    this.props.imgLoaded(true);
  }
  handleImageErrored(){

    this.props.imgLoaded(false);
  }

  render() {
    let urlPic =''
    if(this.props.copy){
      if (this.props.copy.copyList.length != 0){
        for(let i = 0; i<this.props.copy.copyList.length; i++){
          if(this.props.copy.copyList[i].nombre === 'introPic'){
            urlPic = this.props.copy.copyList[i].urlPic;
            break;
          }

        }

      }
    }
    let visual = PopularFunctions.figureOutOpacity(this.props);

    return (
      <div className='intro__foto__container' style = {visual}>
        <img className='intro__foto'
          src= {urlPic}
          onLoad={this.handleImageLoaded.bind(this)}
          onError={this.handleImageErrored.bind(this)}
        ></img>
      </div>
    );
  }
}

const dispatchToProps = (dispatch) =>{

  return{
    getCopy: () => dispatch(actions.getCopy()),
    imgLoaded: (loaded) => dispatch (actions.imgLoaded(loaded)),

  };
};
const stateToProps = (state) => {
  return{
    copy :state.copy,
    section: state.section,
  };
};

export default connect (stateToProps,dispatchToProps)(IntroContainer);
