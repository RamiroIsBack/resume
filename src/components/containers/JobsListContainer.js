import React, { Component } from "react";
import actions from "../../actions";
import JobCss from "../../css"; // eslint-disable-line no-unused-vars
class JobsContainer extends Component {
  render() {
    return <div>mucho jooooobs!!!</div>;
  }
}
export default JobsContainer;
