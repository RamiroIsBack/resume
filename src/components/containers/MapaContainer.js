import React, { Component } from 'react';
import { connect } from 'react-redux';
import {Mapa} from '../presentational';
import { withScriptjs } from 'react-google-maps';
class MapaContainer extends Component {

  render() {
    let mapInfo ={};
    if(this.props.mapInfo){
      mapInfo = this.props.mapInfo;
    }
    return (
      <div>

        <Mapa
          mapInfo = {mapInfo}
          containerElement={<div style={{ height: '260px', maxWidth:'1200px'}} />}
          mapElement={<div style={{ height: '100%' }} />}
        />

      </div>
    );
  }
}


const stateToProps = (state) => {
  return{
    section:state.section,
    copy:state.copy,
  };
};

export default connect (stateToProps,null)(withScriptjs(MapaContainer))
