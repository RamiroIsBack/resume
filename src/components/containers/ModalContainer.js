import React, { Component } from "react";
import { ServiceModal, WorkModal } from "../presentational";

import { connect } from "react-redux";
import actions from "../../actions";

class ModalContainer extends Component {
  moveToCreacionesSection(Name) {
    this.props.moveToCreacionesSection(Name);
    this.props.toggleModal("closeMenuXs");
  }

  toggleModal() {
    this.props.toggleWorkModal("");
  }
  getWorkCopyToShow() {
    let workSelectedName = this.props.section.workModal;
    let completeList = [
      ...this.props.copy.talksList,
      ...this.props.copy.worksList
    ];
    return completeList.find(work => work.nombre === workSelectedName);
  }

  render() {
    var modalShowing = false;

    if (this.props.section) {
      if (
        this.props.section.workModal !== "" ||
        this.props.section.serviceModal !== ""
      ) {
        modalShowing = true;
      }
    }
    // Render nothing if the "show" prop is false
    if (!modalShowing) {
      return null;
    }
    if (this.props.section.workModal !== "") {
      var workToShow = this.getWorkCopyToShow();
      return (
        <div>
          <WorkModal
            show={modalShowing}
            workToShow={workToShow}
            onClose={() => this.props.toggleWorkModal("")}
            mobile={this.props.section.screenSize === "mobile" ? true : false}
          />
        </div>
      );
    } else {
      var serviceToShow = this.props.section.serviceModal;
      return (
        <div>
          <ServiceModal
            show={modalShowing}
            workToShow={serviceToShow}
            onClose={() => this.props.toggleWorkModal("")}
            mobile={this.props.section.screenSize === "mobile" ? true : false}
          />
        </div>
      );
    }
  }
}
const dispatchToProps = dispatch => {
  return {
    toggleWorkModal: modalName => dispatch(actions.toggleWorkModal(modalName))
  };
};

const stateToProps = state => {
  return {
    copy: state.copy,
    section: state.section
  };
};

export default connect(
  stateToProps,
  dispatchToProps
)(ModalContainer);
