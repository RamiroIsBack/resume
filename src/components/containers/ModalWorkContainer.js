import React, { Component } from "react";
import { ModalWork } from "../presentational";

import { connect } from "react-redux";
import actions from "../../actions";

class ModalWorkContainer extends Component {
  moveToCreacionesSection(Name) {
    this.props.moveToCreacionesSection(Name);
    this.props.toggleModal("closeMenuXs");
  }

  toggleModal() {
    this.props.toggleModal("closeMenuXs");
  }
  render() {
    var menuXsShowing = false;
    var creacionesContenidos = {};

    if (this.props.storeModal) {
      menuXsShowing = this.props.storeModal.menuXsShowing;
    }
    // Render nothing if the "show" prop is false
    if (!menuXsShowing) {
      return null;
    }
    for (
      let i = 0;
      i < this.props.storeContenidos.listaContenidos.length;
      i++
    ) {
      if (this.props.storeContenidos.listaContenidos[i].id === "creaciones") {
        creacionesContenidos = this.props.storeContenidos.listaContenidos[i];
      }
    }
    let creacionDB = this.props.storeCreaciones.listaCreaciones;

    var creacionList = [];

    for (var tipo in creacionDB) {
      if (creacionDB.hasOwnProperty(tipo)) {
        creacionList.push({
          nombre: tipo
        });
      }
    }

    return (
      <div>
        <ModalWork
          show={menuXsShowing}
          onSelectCreaciones={this.moveToCreacionesSection.bind(this)}
          onClose={this.toggleModal.bind(this)}
          creacionesContenidos={creacionesContenidos}
          creacionList={creacionList}
        />
      </div>
    );
  }
}
const dispatchToProps = dispatch => {
  return {
    getContenidos: () => dispatch(actions.getContenidos()),
    toggleModal: modalName => dispatch(actions.toggleModal(modalName)),
    getFerias: () => dispatch(actions.getFerias()),
    getCreaciones: () => dispatch(actions.getCreaciones()),
    moveToCreacionesSection: creacionTipo =>
      dispatch(actions.moveToCreacionesSection(creacionTipo)),
    moveToFeriasSection: feriaName =>
      dispatch(actions.moveToFeriasSection(feriaName))
  };
};

const stateToProps = state => {
  return {
    // state is d store in this case for convenction
    // cojo el producto d state(store) y lo paso a props xa cogerlo
    //en state.blabla dices de que reducer quieres info
    //y tu le asignas una key q quieras
    storeContenidos: state.contenidos,
    storeModal: state.modal,
    storeCreaciones: state.creacion,
    storeFerias: state.feria
  };
};
//                                   ****
export default connect(
  stateToProps,
  dispatchToProps
)(ModalWorkContainer);
