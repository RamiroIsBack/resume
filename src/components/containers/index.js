import IntroContainer from './IntroContainer.js';
import SummaryContainer from './SummaryContainer.js';
import ServicesContainer from './ServicesContainer.js';
import SkillsContainer from './SkillsContainer.js';
import TimeLineContainer from './TimeLineContainer.js';
import SideBarContainer from './SideBarContainer.js';
import NombreContainer from './NombreContainer'

export{
  NombreContainer,
  IntroContainer,
  SummaryContainer,
  ServicesContainer,
  SkillsContainer,
  TimeLineContainer,
  SideBarContainer,
};
