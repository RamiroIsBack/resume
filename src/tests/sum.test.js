import React from 'react';
import renderer from 'react-test-renderer';

import {SumContainer} from '../components/containers';

it('sums allright', () => {
  let number = 4;
  const component = renderer.create(
    <SumContainer number = {number} />
  ).toJSON();
  expect (component).toMatchSnapshot();
});
