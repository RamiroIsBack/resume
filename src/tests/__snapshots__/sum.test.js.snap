// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sums allright 1`] = `
<div>
  <h3>
     la suma es 
    9
     
  </h3>
</div>
`;
